x-pack/plugins/security_solution/server/lib/siem_migrations/rules/api/retry.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IKibanaResponse, Logger } from '@kbn/core/server';
+import { buildRouteValidationWithZod } from '@kbn/zod-helpers';
+import { APMTracer } from '@kbn/langchain/server/tracers/apm';
+import { getLangSmithTracer } from '@kbn/langchain/server/tracers/langsmith';
+import {
+  StartRuleMigrationRequestBody,
+  StartRuleMigrationRequestParams,
+  type StartRuleMigrationResponse,
+} from '../../../../../common/siem_migrations/model/api/rules/rule_migration.gen';
+import { SIEM_RULE_MIGRATION_RETRY_PATH } from '../../../../../common/siem_migrations/constants';
+import type { SecuritySolutionPluginRouter } from '../../../../types';
+
+export const registerSiemRuleMigrationsRetryRoute = (
+  router: SecuritySolutionPluginRouter,
+  logger: Logger
+) => {
+  router.versioned
+    .put({
+      path: SIEM_RULE_MIGRATION_RETRY_PATH,
+      access: 'internal',
+      security: { authz: { requiredPrivileges: ['securitySolution'] } },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            params: buildRouteValidationWithZod(StartRuleMigrationRequestParams),
+            body: buildRouteValidationWithZod(StartRuleMigrationRequestBody),
+          },
+        },
+      },
+      async (context, req, res): Promise<IKibanaResponse<StartRuleMigrationResponse>> => {
+        const migrationId = req.params.migration_id;
+        const { langsmith_options: langsmithOptions, connector_id: connectorId } = req.body;
+
+        try {
+          const ctx = await context.resolve([
+            'core',
+            'actions',
+            'alerting',
+            'securitySolution',
+            'licensing',
+          ]);
+          if (!ctx.licensing.license.hasAtLeast('enterprise')) {
+            return res.forbidden({
+              body: 'You must have a trial or enterprise license to use this feature',
```

Comment:
Do we have existing language, some constant, for rejecting features based on licenses that we can re-use here?

Reply:
I have found Endpoint and EA have some translation for license errors, like [this](https://github.com/elastic/kibana/blob/de2d9675b23d7e8a846df938916a0dc06a16a410/x-pack/plugins/security_solution/server/endpoint/services/actions/clients/lib/base_response_actions_client.ts#L102-L107)

I haven't implemented licensing properly so far because this API is still internal, it can only be accessed from within Kibana, and we'll need license checks in the UI anyway.

However, since this API is going to be public in the future, I'll add proper checks to all the routes with i18n 👍